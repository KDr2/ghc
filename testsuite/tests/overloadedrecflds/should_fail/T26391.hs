module T26391 where

import Data.Semigroup (getSum, getProduct)

-- This record update is invalid (no constructor has both 'getSum' and 'getProduct').
--
-- This test makes sure that GHC can handle reporting a good error even when
-- the parent constructors (here, Sum and Product) are out of scope.
a = undefined { getSum = undefined, getProduct = undefined }
