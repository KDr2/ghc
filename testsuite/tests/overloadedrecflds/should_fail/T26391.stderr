T26391.hs:9:5: error: [GHC-14392]
    Invalid record update.
    No constructor in scope has all of the following fields:
      ‘getSum’, ‘getProduct’

